src/App.tsx

```diff
@@ -15,37 +15,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = 'DEFAULT',
+  ALPHABETICAL = 'ALPHABETICAL',
+  LENGTH = 'LENGTH',
+}
+
 export const App: React.FC = () => {
+  const initialGoods = [...goodsFromServer];
+  const [goods, setGoods] = useState<string[]>(initialGoods);
+  const [sortType, setSortType] = useState<SortType>(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortGoods = (type: SortType) => {
+    let sortedGoods = [...initialGoods];
+
+    switch (type) {
+      case SortType.ALPHABETICAL:
+        sortedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SortType.LENGTH:
+        sortedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        sortedGoods = initialGoods;
+        break;
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    setGoods(sortedGoods);
+    setSortType(type);
+  };
+
+  const toggleReverse = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  const resetGoods = () => {
+    setGoods(initialGoods);
```

Comment:
Consider using `setGoods([...initialGoods])` instead of `setGoods(initialGoods)` to ensure that a new array is created, maintaining immutability. This prevents potential side effects if `initialGoods` is modified elsewhere.